db/upgrade.php

```diff
@@ -341,5 +341,35 @@ function xmldb_cms_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2023112100, 'cms');
     }
 
+    if ($oldversion < 2023120101) {
+        // Update query for setting.
+        $sql = "UPDATE {customfield_data} mcd1
+                   SET contextid = sub.mcid
+                  FROM (
+                        SELECT mcd.id mcdid, mcd.contextid mcdcontextid, mc.id mcid
+                          FROM {customfield_data} mcd
+                          JOIN {customfield_field} mcf ON mcf.id = mcd.fieldid
+                          JOIN {customfield_category} mcc ON mcc.id = mcf.categoryid
+                          JOIN {course_modules} mcm ON mcm.instance = mcd.instanceid AND mcm.module = (
+                              SELECT id FROM {modules} WHERE name = 'cms'
+                          )
+                          JOIN {context} mc ON mc.instanceid = mcm.id AND contextlevel = 70
+                         WHERE mcd.id IN (
```

Comment:
why do we need this IN? Can't we just limit it all here like `mcf.type = 'textarea' AND mcc.component = 'mod_cms' AND mcd.contextid != mc.id`

Reply:
Yes, I reviewed and we don't need this subquery here.
I updated the SQL.